.github/scripts/process_commit.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+"""
+This script finds the merger responsible for labeling a PR by a commit SHA. It is used by the workflow in
+'.github/workflows/pr-labels.yml'. If there exists no PR associated with the commit or the PR is properly labeled,
+this script is a no-op.
+
+Note: we ping the merger only, not the reviewers, as the reviewers can sometimes be external to pytorch
+with no labeling responsibility, so we don't want to bother them.
+This script is based on: https://github.com/pytorch/vision/blob/main/.github/process_commit.py
+"""
+
+import sys
+from typing import Any, Set, Tuple
+
+import requests
+
+# For a PR to be properly labeled it should have release notes label and one topic label
+PRIMARY_LEBEL_FILTER = "release notes:"
+SECONDARY_LEBEL_FILTER = "topic:"
+
+def query_pytroch(cmd: str, *, accept: str) -> Any:
+    response = requests.get(f"https://api.github.com/repos/pytorch/pytorch/{cmd}", headers=dict(Accept=accept))
+    return response.json()
+
+
+def get_pr_number(commit_hash: str) -> Any:
+    # See https://docs.github.com/en/rest/reference/repos#list-pull-requests-associated-with-a-commit
+    data = query_pytroch(f"commits/{commit_hash}/pulls", accept="application/vnd.github.groot-preview+json")
+    if not data:
+        return None
+    return data[0]["number"]
+
+
+def get_pr_merger_and_labels(pr_number: int) -> Tuple[str, Set[str]]:
+    # See https://docs.github.com/en/rest/reference/pulls#get-a-pull-request
+    data = query_pytroch(f"pulls/{pr_number}", accept="application/vnd.github.v3+json")
+    merger = data["merged_by"]["login"]
+    labels = {label["name"] for label in data["labels"]}
+    return merger, labels
+
+
+if __name__ == "__main__":
+    commit_hash = sys.argv[1]
+    pr_number = get_pr_number(commit_hash)
+
+    if not pr_number:
+        sys.exit(0)
+
+    merger, labels = get_pr_merger_and_labels(pr_number)
+    response = query_pytroch("labels", accept="application/json")
+    response_labels = list(map(lambda x: str(x["name"]), response.json()))
+    primary_labels = set(filter(lambda x: x.startswith(PRIMARY_LEBEL_FILTER), response_labels))
+    secondary_labels = set(filter(lambda x: x.startswith(SECONDARY_LEBEL_FILTER), response_labels))
+
+    is_properly_labeled = bool(primary_labels.intersection(labels) and secondary_labels.intersection(labels))
```

Comment:
@malfet do we wanna mandate people to add labels upfront too?

Reply:
> Also, before we merge this PR we should actually add all the release notes category and topic labels. As a way to be transparent and avoid confusion, should we have the `facebook-github-bot` link the issue containing available release categories and labels in its comment on each PR?

@anjali411 and @malfet Acknoweldge before merging, will add necessary labels

Reply:
> @malfet do we wanna mandate people to add labels upfront too?
@anjali411 Can you please explain the definition of mandate, prevent from merging ? 

What is currently implemented is simply pinging people, if the labels are not applied, perhaps we should implement this as a first stage. And enforcing this policy can be a followup PR. I guess we will have to notify people that there is this new policy in place etc...

Reply:
sorry I meant add topic labels upfront